src/components/NcActionButton/NcActionButton.vue

```diff
@@ -269,6 +381,34 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+
+		/**
+		 * The button's behavior, by default the button acts like a normal button with optional toggle button behavior if `modelValue` is `true` or `false`
+		 * But you can also set to checkbox button behavior with tri-state or radio button like behavior.
+		 */
+		modelBehavior: {
+			type: String,
+			default: 'button',
+			validator: (behavior) => ['checkbox', 'radio', 'button'].includes(behavior),
+		},
+
+		/**
+		 * The buttons state if `modelBehavior` is 'checkbox' or 'radio' (meaning if it is pressed / selected)
+		 * @type {boolean|any} Either boolean for checkbox and toggle button behavior or `radioValue` for radio behavior
+		 */
+		// eslint-disable-next-line vue/require-prop-types
```

Comment:
Is there a proper way to set `type` to `any`?

Reply:
I think it shouldn't be `any`. Like in native radio/checkboxes. Either `boolean` or `string`.

Reply:
Ok so like NcCheckboxRadioSwitch